spec/requests/admin/configs_spec.rb

```diff
@@ -807,15 +807,17 @@
         end
 
         it "unsets appropriate SMTP config, and apply default value if applicable" do
-          Settings::SMTP.address = "smtp.example.com"
-          Settings::SMTP.port = 12_345
+          default_address = ApplicationConfig["SMTP_ADDRESS"].presence || nil
+          default_port = ApplicationConfig["SMTP_PORT"].presence.to_i || 25
           expected_handle = { "address" => "", "port" => "" }
+
           post admin_settings_smtp_settings_path, params: {
             settings_smtp: expected_handle,
             confirmation: confirmation_message
           }
-          expect(Settings::SMTP.address).to eq(nil)
-          expect(Settings::SMTP.port).to eq(25)
+
+          expect(Settings::SMTP.address).to eq(default_address)
+          expect(Settings::SMTP.port).to eq(default_port)
```

Comment:
@citizen428 @rhymes
This expectation on `Settings::SMTP.port` keeps failing CI (although it passes consistently on local):

```
 1) /admin/customization/config POST /admin/customization/config when admin has full permissions including super SMTP configs unsets appropriate SMTP config, and apply default value if applicable
     Failure/Error: expect(Settings::SMTP.port).to eq(default_port)
       expected: 0
            got: 25
```

On local, if I say `expect(Settings::SMTP.port).to eq(25)` (as the CI test seems to want), it fails locally. This is why I set the expectation to expect `default_port`. But this approach fails CI.

How do I make this spec more robust to pass locally and on Travis? 😢  😓  Thank you!

Reply:
So, `Settings::SMTP.port` defaults to `ApplicationConfig["SMTP_PORT"].presence || 25` so it's always 25 unless set to a different value. The reason why you're getting 0 here is because the test does `.to_i` which becomes 0 which is `trueish` in Ruby.  So `0 || 25` will yield 0.

```ruby
default_port = (ApplicationConfig["SMTP_PORT"].presence || 25).to_i
```

is probably what you're looking for.

BTW `ApplicationConfig["SMTP_ADDRESS"].presence || nil` is redundant as `.presence` will return nil if it's not present

Hopefully this help!